pos_exercises/static/src/model.js

```diff
@@ -0,0 +1,139 @@
+/** @odoo-module **/
+
+import models from "point_of_sale.models";
+
+import utils from "web.utils";
+import field_utils from "web.field_utils";
+
+var round_pr = utils.round_precision;
+
+models.load_fields("res.partner", ["title"]);
+models.load_fields("pos.order.line", ["bundle_id"]);
+
+models.load_models([{
+    model: "res.partner.title",
+    domain: [],
+    fields: ["name"],
+    loaded: function (pos, partnerTitles) {
+        pos.partnerTitles = partnerTitles;
+    }
+}]);
+
+models.load_models([{
+    model: "pos.bundle",
+    domain: [],
+    fields: ["name", "product_ids"],
+    loaded: function (pos, posBundles) {
+        pos.posBundles = posBundles;
+        pos.posBundleById = Object.fromEntries(posBundles.map(obj => [obj.id, obj]));
+    }
+}]);
+
+var order_line_super = models.Orderline.prototype;
+models.Orderline = models.Orderline.extend({
+
+    initialize: function () {
+        this.bundle_id = null;
+        order_line_super.initialize.apply(this, arguments);
+    },
+
+    init_from_JSON: function(json) {
+        order_line_super.init_from_JSON.apply(this, arguments);
+        this.bundle_id = json.bundle_id || null;
+    },
+
+    export_as_JSON: function() {
+        var json = order_line_super.export_as_JSON.apply(this,arguments);
+        if (this.bundle_id){
+            json.bundle_id = this.bundle_id.id;
+        }
+        return json;
+    },
+
+    setBundleId: function (bundle_id) {
+        this.bundle_id = bundle_id;
+        this.trigger('change', this);
+    },
+
+    export_for_printing: function () {
+        return {
+            ...order_line_super.export_for_printing.apply(this, arguments),
+            bundle_id: this.bundle_id,
+        }
+    },
+
+    // Inherit "OverWrite"
+    set_quantity: function(quantity, keep_price){
```

Comment:
- you did not need to replace the entire method here. You could've done something like the following:

```js
set_quantity: function(quantity) {
    if (quantity === 'remove' && this.bundle_id){
        const lines_ids = this.order.get_orderlines();
        const new_lines_ids = lines_ids.filter((line_id) => line_id.bundle_id && line_id.bundle_id.id == bundle_id.id);
        this.order.orderlines.remove(new_lines_ids);
    }

    return _order_super.set_quantity.apply(this, arguments);
}
```

- It is a better idea to implement this logic in the ProductScreen component actually. (`ProductScreen.prototype._setValue()`) since you want the logic to be triggered based on user interaction. No need to implement it on this level